spec/system/candidate_entering_academic_qualifications_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+describe 'A candidate adding a Degree' do
+  context 'who successfully enters their details' do
+    before do
+      visit '/degrees/new'
+
+      fill_in_degree_details
+
+      click_on t('application_form.save_and_continue')
+    end
+
+    it 'sees a summary of those details' do
+      expect(page).to have_content('BA')
+    end
+
+    context 'and wishes to amend their details' do
+      it 'can go back and edit them' do
+        visit '/check_your_answers'
+
+        find('#change-degree').click
+        expect(page).to have_field('Type of degree', with: 'BA')
+      end
+    end
+  end
+
+  context 'who leaves out a required field' do
+    before do
+      visit '/contact_details/new'
+    end
+
+    it 'sees an error summary' do
+      click_on t('application_form.save_and_continue')
+
+      expect(page).to have_content('There is a problem')
+    end
+  end
+
+private
+
+  def fill_in_degree_details
```

Comment:
For personal details, we extracted this method into a support module (4eba8c031f24fb8bdf0f1cfb18b9628b00cb6ceb). For contact details and here, we haven't done that.

Which approach is better? I would like to be consistent.

Reply:
I am more slightly in favour to keep it in private method:
- it is clear what are the values to fill the fields
- for now, it is the only test using it (and probably it will be for a while)

Reply:
I have moved the equivalent method for `contact_details` into a module during work on https://trello.com/c/FSbqIXPu/650-add-edit-academic-qualification-non-degree because this allows it to be reused in the end-to-end spec `candidate_completing_an_application_spec.rb` — which I think is not updated in this branch?